### PR TITLE
Make ILogger strict

### PR DESCRIPTION
**Pull request by rullzer**
* Make implementations strict
* Add scalar typehints

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: lib/public/ILogger.php [hunk 1]
```diff
@@ -125,7 +126,7 @@ public function debug($message, array $context = array());
 	 * @return mixed
 	 * @since 7.0.0
 	 */
-	public function log($level, $message, array $context = array());
+	public function log($level, string $message, array $context = []);
```
Comment (MorrisJobke) [inline lib/public/ILogger.php]: Why is level not hinted as integer?

File: lib/public/ILogger.php [hunk 2]
```diff
@@ -143,5 +144,5 @@ public function log($level, $message, array $context = array());
 	 * @return void
 	 * @since 8.2.0
 	 */
-	public function logException($exception, array $context = array());
+	public function logException($exception, array $context = []);
```
Comment (MorrisJobke) [inline lib/public/ILogger.php]: \Error should work here, right?

File: lib/public/ILogger.php (hunk 1 above)
Comment (rullzer) [reply lib/public/ILogger.php]: Because the phpdoc says it is mixed.

File: lib/public/ILogger.php (hunk 2 above)
Comment (rullzer) [reply lib/public/ILogger.php]: Ah mmm even just Throwable I guess as Exception extends throwable.

Comment (ChristophWurst) [reply lib/public/ILogger.php]: > Ah mmm even just Throwable I guess as Exception extends throwable

Correct. https://secure.php.net/manual/en/class.throwable.php says

> Throwable is the base interface for any object that can be thrown via a throw statement in PHP 7, including Error and Exception.

**Review by MorrisJobke — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**